src/operator/channel_op_common.h

```diff
@@ -101,6 +101,42 @@ void Split(const mshadow::Tensor<xpu, dim, DType> &input,
     split_helper<xpu, dim, dim-1>(input, output, dimension, req);
   }
 }
+
+template<typename cpu, int dim, typename DType>
```

Comment:
Current template for `split` and `concatenate` is general for cpu and gpu. This Split_2D only works for CPU. Do you plan to add a GPU version as well? Otherwise, it seems to break the existing pattern.